store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedWhileLoopAndJoinTest.java

```diff
@@ -0,0 +1,280 @@
+import static uk.gov.gchq.gaffer.user.User.UNKNOWN_USER_ID;
+
+public class FederatedWhileLoopAndJoinTest {
+    @BeforeEach
+    public void before() {
+        resetForFederatedTests();
+    }
+
+    @Test
+    void shouldWhileLoopOperationWithDistinctOperations() throws Exception {
+        //given
+        final FederatedStore federatedStore = new FederatedStore();
+        final FederatedStoreProperties properties = new FederatedStoreProperties();
+        federatedStore.initialise(GRAPH_ID_TEST_FEDERATED_STORE, new Schema(), properties);
+
+        addGraphToAccumuloStore(federatedStore, GRAPH_ID_ACCUMULO, true, basicEntitySchema());
+
+        //1st Add Elements is distinct.
+        federatedStore.execute(new AddElements.Builder()
+                .input(entityBasic())
+                .build(), contextTestUser());
+
+        //2nd Add Elements is distinct.
+        final Operation operation = new While.Builder()
+                .operation(new
+                        AddElements.Builder()
+                        .input(entityBasic())
+                        .build())
+                .conditional(o -> true)
+                .maxRepeats(2)
+                .build();
+
+        //when
+        federatedStore.execute(operation, contextTestUser());
+
+        //then
+        assertThat(federatedStore.execute(new GetAllElements(), contextTestUser()))
+                .containsExactly(new Entity.Builder()
+                        .group(GROUP_BASIC_ENTITY)
+                        .vertex(BASIC_VERTEX)
+                        .property(PROPERTY_1, 3)
+                        .build());
+    }
+
+    @Test
+    void shouldWhileLoopOperationWithJoinOperation() throws Exception {
+        //given
+        final FederatedStoreProperties properties = new FederatedStoreProperties();
+        properties.set(CacheProperties.CACHE_SERVICE_CLASS, HashMapCacheService.class.getName());
+
+        //Use graph due to hooks needed.
+        final Graph federated = new Graph.Builder()
+                .addSchema(new Schema())
+                .config(new GraphConfig(GRAPH_ID_TEST_FEDERATED_STORE))
+                .storeProperties(properties)
+                .build();
+
+
+        final Schema schema = basicEntitySchema();
+        final StoreProperties storeProperties = FederatedStoreTestUtil.loadAccumuloStoreProperties(FederatedStoreTestUtil.ACCUMULO_STORE_SINGLE_USE_PROPERTIES);
+        final Context context = new Context(new User(UNKNOWN_USER_ID));
+        federated.execute(OperationChain.wrap(
+                new AddGraph.Builder()
+                        .graphId(GRAPH_ID_ACCUMULO)
+                        .storeProperties(storeProperties)
+                        .schema(schema)
+                        .parentSchemaIds(null)
+                        .parentPropertiesId(null)
+                        .isPublic(true)
+                        .build()), context);
+
+        final Operation operation = JSONSerialiser.deserialise("{\n" +
```

Comment:
Might be easier to read as Java objects rather than string

Reply:
Probably, the operation came from another team as JSON.